lib/rubocop/cop/rails/environment_comparison.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop checks that Rails.env is compared using `.production?`-like
+      # methods instead of equality against a string or symbol.
+      #
+      # @example
+      #   # bad
+      #   Rails.env == 'production'
+      #
+      #   # bad, always returns false
+      #   Rails.env == :test
+      #
+      #   # good
+      #   Rails.env.production?
+      class EnvironmentComparison < Cop
+        MSG = 'Favor `Rails.env.%s?` over `Rails.env == %s`.'.freeze
+
+        def_node_matcher :environment_comparison?, <<-PATTERN
+          (send
+            (send (const {nil? cbase} :Rails) :env)
+            :==
+            {str sym}
+          )
+        PATTERN
+
+        def on_send(node)
+          return unless environment_comparison?(node)
+          env = node.children.last
```

Comment:
I'd replace all the usages of `children` in the code with splat destructuring of nodes. The performance is better and it's more clear what exactly are you extracting. Generally we avoid `children` everywhere in the codebase.

Reply:
Yes, yes. I'll add an internal investigation for it. 😉